.github/workflows/copilot-setup-steps.yml

```
@@ -0,0 +1,62 @@
+name: "Copilot Setup Steps"
+
+# This workflow runs when a GitHub Copilot Coding Agent starts and is used to set up the development environment for the agent:
+# https://docs.github.com/en/copilot/how-tos/agents/copilot-coding-agent/customizing-the-development-environment-for-copilot-coding-agent
+
+# Automatically run the setup steps when they are changed to allow for easy validation, and
+# allow manual testing through the repository's "Actions" tab
+on:
+  workflow_dispatch:
+  push:
+    paths:
+      - .github/workflows/copilot-setup-steps.yml
+  pull_request:
+    paths:
+      - .github/workflows/copilot-setup-steps.yml
+
+jobs:
+  # The job MUST be called `copilot-setup-steps` or it will not be picked up by Copilot.
+  copilot-setup-steps:
+    runs-on: ubuntu-latest
+
+    # Set the permissions to the lowest permissions possible needed for your steps.
+    # Copilot will be given its own token for its operations.
+    permissions:
+      # If you want to clone the repository as part of your setup steps, for example to install dependencies, you'll need the `contents: read` permission. If you don't clone the repository in your setup steps, Copilot will do this for you automatically after the steps complete.
+      contents: read
+
+    # You can define any steps you want, and they will run before the agent starts.
+    # If you do not check out your code, Copilot will do this for you.
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Install PowerShell modules for AVM validation
+        uses: azure/powershell@v2
+        with:
+          azPSVersion: "latest"
+          inlineScript: |
+            # Install required PowerShell modules for AVM testing
+            Install-Module -Name Pester -Force -AllowClobber -Scope CurrentUser -RequiredVersion '5.6.1'
+            Install-Module -Name PSRule.Rules.Azure -Force -AllowClobber -Scope CurrentUser
+
+            # Verify installations
+            Get-Module -ListAvailable Az, Pester, PSRule.Rules.Azure
```

Can probably replace this with the [avm-SetEnvironment](https://github.com/Azure/bicep-registry-modules/blob/main/.github/actions/templates/avm-setEnvironment/action.yml) action template we have and embed in all workflows, so its just one lot of code we have to maintain 👍


```suggestion
      - name: Set environment
        uses: ./.github/actions/templates/avm-setEnvironment
```